internal/rbd/encryption_test.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2023 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/ceph/ceph-csi/internal/util"
+)
+
+func TestParseEncryptionOpts(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		testName     string
+		volOptions   map[string]string
+		fallbackType util.EncryptionType
+		expectedKMS  string
+		expectedEnc  util.EncryptionType
+		expectedErr  string
```

Comment:
Instead of a string, you can just check error is expected or not with a bool value here.

Reply:
Thanks for the review, Madhu.
Updated the PR with mentioned changes.